semgrep-core/src/tainting/Dataflow_tainting.ml

```diff
@@ -409,6 +409,11 @@ let check_tainted_tok env tok =
   | _ :: _ -> (Taints.empty, env.lval_env)
   | [] ->
       let taints = source_pms |> taints_of_matches in
+      (* Empty because we have no base, but we want to filter the taints we got from
```

Comment:
Totally fair if this comment just needs a lot of context, but from this snippet it's hard to understand. Main sources of confusion are probably "because we have no base"

Reply:
> a tainted token, which the Dot case was degenerating to

This from the PR is helpful, could use it in the code